haystack/tracing/tracer.py

```diff
@@ -103,3 +109,42 @@ def disable_tracing() -> None:
 def is_tracing_enabled() -> bool:
     """Return whether tracing is enabled."""
     return not isinstance(_tracer, NullTracer)
+
+
+def auto_enable_tracing() -> None:
+    """Auto-enable the right tracing backend.
+
+    This behavior can be disabled by setting the environment variable `HAYSTACK_AUTO_TRACE_ENABLED` to `false`.
+    Note that it will only work correctly if tracing was configured _before_ Haystack is imported.
+    """
+    if os.getenv(HAYSTACK_AUTO_TRACE_ENABLED_ENV_VAR, "true").lower() == "false":
+        logger.info("Tracing disabled via '%s'", HAYSTACK_AUTO_TRACE_ENABLED_ENV_VAR)
+        return
+
+    tracer = _auto_configured_opentelemetry_tracer()
+    if tracer:
+        enable_tracing(tracer)
+        logger.info("Tracing enabled via '%s'", tracer.__class__.__name__)
+
+
+def _auto_configured_opentelemetry_tracer() -> Optional[Tracer]:
+    # we implement this here and not in the `opentelemetry` module to avoid import warnings when OpenTelemetry is not
+    # installed
+    try:
+        import opentelemetry.trace
+
+        # the safest way to check if tracing is enabled is to try to start a span and see if it's a no-op span
+        # alternatively we could of course check `opentelemetry.trace._TRACER_PROVIDER`
+        # but that's not part of the public API and could change in the future
+        with opentelemetry.trace.get_tracer("haystack").start_as_current_span("haystack.tracing.auto_enable") as span:
+            if isinstance(span, opentelemetry.trace.NonRecordingSpan):
+                return
+            else:
+                from haystack.tracing.opentelemetry import OpenTelemetryTracer
+
+                return enable_tracing(OpenTelemetryTracer(opentelemetry.trace.get_tracer("haystack")))
```

Comment:
should we also automatically instrument third-party libraries (e.g. OpenAI, urllib)?

This usually works like the following:

```python
from opentelemetry.instrumentation.urllib3 import URLLib3Instrumentor

URLLib3Instrumentor().instrument()
```

Issue is that opentelemetry will log a warning if these libraries are already instrumented. Depending on how they configure things this is done automatically anyway by OpenTelemetry so I'd skip this for now and only add if required?